File: go/easy/3718_smallest_missing_multiple_of_k.go
import "math"

func missingMultiple(nums []int, k int) int {
    set := make(map[int] struct{})
    for _, n := range nums {
        set[n] = struct{}{}
    }

    for i := 1; i <= math.MaxInt32; i++ {
        if _, ok := set[i]; !ok && i % k == 0 {
            return i
        }
    }

    return -1
}

File: rust/easy/3718_smallest_missing_multiple_of_k.rs
use std::collections::HashSet;

impl Solution {
    pub fn missing_multiple(nums: Vec<i32>, k: i32) -> i32 {
        let set: HashSet<i32> = HashSet::from_iter(nums.iter().cloned());
        for i in 1..=i32::MAX {
            let i = i as i32;
            if !set.contains(&i) && i % k == 0 {
                return i;
            }
        }
        -1
    }
}
